WcaOnRails/app/assets/javascripts/application.js

```diff
@@ -46,25 +46,38 @@ wca.cancelPendingAjaxAndAjax = function(id, options) {
   return wca._pendingAjaxById[id];
 };
 
+function goodDate(element) {
+  element.parent().siblings('p').removeClass('alert alert-danger');
+  element.removeClass('alert-danger');
+}
+
+function badDate(element) {
+  element.parent().siblings('p').addClass('alert alert-danger');
+  element.parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
+  element.addClass('alert-danger');
+}
+
 $(function() {
   $('.dropdown-toggle').dropdownHover();
   $('form.are-you-sure').areYouSure();
 
   // Copied (and modified by jfly) from
-  //  https://github.com/zpaulovics/datetimepicker-rails
+  // https://github.com/zpaulovics/datetimepicker-rails
   $('.date_picker.form-control, .datetime_picker.form-control').datetimepicker({
     useStrict: true, keepInvalid: true, useCurrent: false
   });
 
-  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.error', function(e){
-    $(this).parent().siblings('p').addClass('alert alert-danger');
-    $(this).parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
-    $(this).addClass('alert-danger');
-  });
+  $('.date_picker.form-control').on('blur', function(){
```

Comment:
change doesn't do anything, and input gets too annoying because it flashes
all the time :P

On Mon, Feb 8, 2016 at 5:33 PM, Jeremy Fleischman <notifications@github.com>
wrote:

> In WcaOnRails/app/assets/javascripts/application.js
> <https://github.com/cubing/worldcubeassociation.org/pull/379#discussion_r52196041>
> :
>
> >    $('.date_picker.form-control, .datetime_picker.form-control').datetimepicker({
> >      useStrict: true, keepInvalid: true, useCurrent: false
> >    });
> >
> > -  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.error', function(e){
> > -    $(this).parent().siblings('p').addClass('alert alert-danger');
> > -    $(this).parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
> > -    $(this).addClass('alert-danger');
> > -  });
> > +  $('.date_picker.form-control').on('blur', function(){
>
> Again, blur doesn't seem like right event to be listening to here. What
> about change or even input? Also, nitpick on your spacing here:
> function(){ -> function() {
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cubing/worldcubeassociation.org/pull/379/files#r52196041>
> .
>